packages/client/src/components/TransactionList.js

```diff
@@ -1,78 +1,47 @@
 import React from "react";
-import { useModalContext } from "../contexts";
-import { getFlowscanUrlForTransaction, shortenAddr } from "../utils";
+import { useModalContext } from "contexts";
 import TransactionStatusIcon from "./TransactionStatusIcon";
 
-const FALLBACK_RECIPIENT_NAME = "No name";
+const maxWidths = {
+  index: 50,
+  status: 140,
+  date: 160,
+  action: 70,
+};
 
-function getDisplayStringForTokenTransfer(tokenTransfer) {
-  if (!tokenTransfer) {
-    return "";
-  }
+const Column = ({ className = "", children, style = {} }) => (
+  <div 
+    className={`column is-flex is-align-items-center ${className}`} 
+    style={style}>
+    {children}
+  </div>
+);
 
-  const amount = parseFloat(tokenTransfer.amount.value ?? "0");
-  const token = tokenTransfer?.amount?.token?.id
-    .split(".")[2]
-    .replace("Token", "")
-    .toUpperCase();
-
-  return `${amount} ${token}`;
-}
-
-function TransactionDetails({ safeOwners, transaction, onClose }) {
-  const renderConfirmation = (address) => {
-    return (
-      <div
-        key={address}
-        className="has-background-black rounded-sm has-text-white is-flex is-align-items-center p-4 mr-2"
-      >
-        {shortenAddr(address)}
-      </div>
-    );
+function getUserFriendlyIntent(intent) {
+  const friendlyNameMap = {
+    "FlowToken": "FLOW",
   };
+  const newWords = intent.split(" ").map(word => {
+    const isNumber = !Number.isNaN(parseFloat(word));
+    // handle numbers but avoid addresses
+    if (isNumber && !word.startsWith('0x')) {
+      return parseFloat(word);
+    } 
+    // handle class names
+    if (word.startsWith('A.')) {
+      const chunks = word.split('.');
+      const cadenceClassName = chunks[chunks.length - 2];
+      return friendlyNameMap[cadenceClassName] ?? cadenceClassName;
+    }
 
-  const renderConfirmationsRow = () => {
-    return (
-      <div className="pt-4">
-        <div className="is-flex is-align-items-center is-justify-content-space-between mb-3 has-text-grey">
-          <div>Confirmations</div>
-          <div>
-            {transaction.authorizers.length} out of {safeOwners.length}
-          </div>
-        </div>
-        <div
-          className="pb-5 is-flex is-align-items-center is-fullwidth"
-          style={{ overflow: "auto" }}
-        >
-          {transaction.authorizers.map(({ address }) =>
-            renderConfirmation(address)
-          )}
-        </div>
-      </div>
-    );
-  };
+    return word;
+  });
 
-  const renderRow = (label, value, labelIsGrey) => {
-    return (
-      <div className="is-flex is-align-items-center is-justify-content-space-between py-5 border-light-top">
-        <label className="has-text-grey">{label}</label>
-        <div className={`has-text-${labelIsGrey ? "grey" : "black"}`}>
-          {value}
-        </div>
-      </div>
-    );
-  };
+  return newWords.join(" ");
+}
 
+const TransactionDetails = ({ transaction, onClose }) => {
```

Comment:
Can we move this into a separate file?

Reply:
I will be in a follow up PR! I have a ticket for just the detail modal